### DISCUSSION TITLE
Cannot read properties of null

### DISCUSSION THREAD

**Discussion by livesayjesse** [NUWCDIVNPT/stig-manager#741] [Q&A]
Hello. We recently got STIG Manager working and now we are fine tuning our collections, automation, etc. When we try to export ckl of assets (with the bundle checklist together in a single file). We get this error message:
![image](https://user-images.githubusercontent.com/82839865/185651930-1823965d-2c9f-497e-b01b-293548aebb98.png)

**Comment by livesayjesse** [comment]
I have noticed that exporting other assets is fine. This seems to only happen when the Server 2019 checklist is part fo the asset's checklist. If I export an asset with a Server 2019 and do not select the "Export multi-STIG ckl" button. I get all the other checklist except for Server 2019.

**Reply by cd-rite** [reply]
Hi @livesayjesse The logs that @csmig asks for would provide a more definitive answer, but I would suggest you increase the size of your MySQL `sort_buffer_size` variable. Those Windows Server STIG checklists can sometimes have very large detail/comment text fields (Especially if you are using the Evaluate-STIG utility), and we have had occasional reports of errors when generating them.  Those issues were all resolved when the deployers increased the size of that buffer. 
**I would suggest setting the global sort_buffer_size  for the stigman db to 64mb or higher**, but you can experiment with it to see what works for you. Please let us know if that resolves your issue! Thanks!

More info here: [https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_sort_buffer_size](https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_sort_buffer_size)

I will adjust our Documentation to make this recommendation prominent.

**Comment by csmig** [comment]
I'm not able to reproduce this error when exporting an asset with an empty (not reviewed) Server 2019 STIG, V2R4. So answers to a few questions might help:
- Which version of STIG Manager? If running from source, which commit? If from our published Docker Hub image, which tag? In general, we only support our latest code and the CKL/XCCDF export code is being actively worked.
- What is the state of the checkbox "Use streaming API"?
- Are there error (`level: 1`) messages logged by the API as it handles the export requests?
  - If you're not using the streaming API, the request for the multi-STIG CKL will be `GET /assets/{assetId}/checklists`. The request for the mono-STIG CKL for Server 2019 will be `GET /assets/{assetId}/checklists/Windows_Server_2019_STIG/latest?format=ckl`
  - If you are using the streaming API and running the current code, the request will be `POST /collections/{collectionId}/archive/ckl`. For code prior to this week, the request is made to `POST /collections/{collectionId}/ckl-archive`.

Thanks for any further info.
